public/js/search.js

```diff
@@ -0,0 +1,66 @@
+$(function(config){
+  var algolia = algoliasearch(config.applicationId, config.apiKey);
+  var index = algolia.initIndex(config.indexName);
+
+  hideSearch();
+
+  $('.clear-search img, .search-results-container img').on('click', function() {
```

Comment:
I'd maybe just put a `.clear-search` class on the X image in the results container, and use the `clear-search` class that's on the input button as your click handler, rather than putting the click handler on the image within it. e.g. this would be refactored to just:

`$('.clear-search').on('click', function() {`